src/github_runner_manager/reactive/consumer.py

```diff
@@ -85,27 +110,31 @@ def consume(
 
     Raises:
         JobError: If the job details are invalid.
+        QueueError: If an error when communicating with the queue occurs.
     """
-    with Connection(queue_config.mongodb_uri) as conn:
-        with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
-            with signal_handler(signal.SIGTERM):
-                msg = simple_queue.get(block=True)
-                try:
-                    job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
-                except ValidationError as exc:
-                    msg.reject(requeue=True)
-                    raise JobError(f"Invalid job details: {msg.payload}") from exc
-                logger.info(
-                    "Received job with labels %s and job_url %s",
-                    job_details.labels,
-                    job_details.url,
-                )
-                _spawn_runner(
-                    runner_manager=runner_manager,
-                    job_url=job_details.url,
-                    msg=msg,
-                    github_client=github_client,
-                )
+    try:
+        with Connection(queue_config.mongodb_uri) as conn:
+            with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
+                with signal_handler(signal.SIGTERM):
+                    msg = simple_queue.get(block=True)
+                    try:
+                        job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
+                    except ValidationError as exc:
+                        msg.reject(requeue=True)
```

Comment:
Wouldn't we get stuck here? I.e., how would we get to the point where this exception wasn't raised if we consume this message again?

Reply:
This is a valid point. I can change this to set `requeue=False`. However, we need to consider in the future that if we have different consumer versions and the JobDetails change, one consumer might consume a job and the other might not, and dropping the message would turn the job lost.

Reply:
Changed to `requeue=False`